api/server.ts

```diff
@@ -0,0 +1,55 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import child_process from 'child_process';
+import fs from 'fs';
+const app = express();
+
+const TEMPLATE = 'dist/__app.html';
+const PORT = 5000;
+
+// helper function to compile gator
+// this could be vastly simplified with some changes to the gator executable
+function gatorc(content: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    fs.writeFile('/tmp/gator.lgl', content, () => {});
+    child_process.execFile('gatorc', ['/tmp/gator.lgl'], (e, stdout, stderr) => {
```

Comment:
Currently, gator compilation is just done on the server side via executing `gatorc`. I implemented this for now because it was a lot of easier than `js_of_ocaml`, but we should continue trying to figure out how to get `js_of_ocaml` to actually work.

Reply:
This assumes you have `gatorc` on your `$PATH`.